util/publicdns/publicdns.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package publicdns contains mapping and helpers for working with
+// public DNS providers
+package publicdns
+
+import "inet.af/netaddr"
+
+var KnownDoH = map[netaddr.IP]string{} // 8.8.8.8 => "https://..."
+
+var DohIPsOfBase = map[string][]netaddr.IP{}
+
+func addDoH(ipStr, base string) {
+	ip := netaddr.MustParseIP(ipStr)
+	KnownDoH[ip] = base
+	DohIPsOfBase[base] = append(DohIPsOfBase[base], ip)
+}
+
+func DohV6(base string) (ip netaddr.IP, ok bool) {
```

Comment:
exported symbols need proper docs (higher bar for exported stuff than package internals) in normal Go complete sentence style (https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences)

Also, this would be `DoH` per https://github.com/golang/go/wiki/CodeReviewComments#initialisms probably.